src/directives/scaleselector.js

```diff
@@ -243,6 +247,7 @@ ngeo.ScaleselectorController.prototype.handleResolutionChange_ = function(e) {
  */
 ngeo.ScaleselectorController.prototype.handleViewChange_ = function(e) {
   this.registerResolutionChangeListener_();
+  this.handleResolutionChange_(null);
```

Comment:
I think that was already needed before, but the problem was not detected.

As soon as the view has been changed, the current scale could be different.
When the view changes, the function ``registerResolutionChangeListener_`` is called and registers a listener on the  ``change:resolution`` event but the event is not fired and ``this.currentScale`` is not initialized with the new current scale belonging to the new view.